lib/kramdown/converter/syntax_highlighter/rouge.rb

```diff
@@ -27,7 +27,16 @@ def self.call(converter, text, lang, type, call_opts)
       lexer = ::Rouge::Lexer.find_fancy(lang || opts[:default_lang], text)
       return nil if opts[:disable] || !lexer
 
-      formatter = (opts.fetch(:formatter, ::Rouge::Formatters::HTML)).new(opts)
+      begin
+        # Rouge::Formatters::HTMLLegacy is available on Rouge 2.0 or later
+        formatter_class = ::Rouge::Formatters::HTMLLegacy
+        opts[:css_class] ||= 'highlight'
+      rescue NameError
+        # Fallback to Rouge 1
+        formatter_class = ::Rouge::Formatters::HTML
+      end
```

Comment:
This `begin ... rescue ... end` block should go outside the method and inside the `begin ... rescue` block where we test for availability.